core/domain/exp_services.py

```diff
@@ -1958,16 +2167,30 @@ def are_changes_mergeable(exp_id, frontend_version, change_list):
                 if change.state_name in state_names_of_renamed_states:
                     old_state_name = (
                         state_names_of_renamed_states[change.state_name])
-                    new_state_name = (
-                        state_names_of_renamed_states[change.state_name])
-                if change.state_name in old_to_new_state_names:
-                    new_state_name = old_to_new_state_names[old_state_name]
+                if old_state_name in old_to_new_state_names:
+                    # Here we will send the changelist, frontend_version,
+                    # backend_version and exploration to the admin, so
+                    # that the changes related to state renames can be
+                    # reviewed and the proper conditions can be written
+                    # to handle those cases.
```

Comment:
Can you leave a comment here and elsewhere explaining a few known situations where it can be complicated to handle? [This will help reviewers to quickly understand the situation.]

Reply:
I have explained the scenario here in this comment https://github.com/oppia/oppia/pull/13233/files#r663565302 that why are we doing this. I hope it's clear.